pkg/broker/handler/pool_test.go

```diff
@@ -138,3 +143,22 @@ func GetFreePort() (int, error) {
 	defer l.Close()
 	return l.Addr().(*net.TCPAddr).Port, nil
 }
+
+type FakeAuthenticationCheck struct {
+	authType authcheck.AuthType
+	noError  bool
+}
+
+func NewFakeAuthenticationCheck(authType authcheck.AuthType, noError bool) authcheck.AuthenticationCheck {
+	return &FakeAuthenticationCheck{
+		authType: authType,
+		noError:  noError,
+	}
+}
```

Comment:
nit: you can save this method and directly use `&FakeAuthenticationCheck{}`. Also is there any need to use `FakeXxx` instead of `fakeXxx`?